infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -218,12 +211,11 @@ private void initializeLogMetadata() {
                     logMetadata.update(logEntry, true);
                 }
             } finally {
-                segment.close();
+                segmentSupervisor.close(segment);
```

Comment:
You can make this a `try-with-resources`

Reply:
I can't because segmentSupervisor contains many segments, I have close a particular one here.
Also, I can't just execute segment.close() I have to remove the segment from the list